vmdk_plugin/utils/fs/fs_windows.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is the filesystem interface for mounting volumes on a Windows guest.
+
+package fs
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+const (
+	// FstypeDefault specifies the default FS to be used when not specified by the user.
+	FstypeDefault = ntfs
+
+	maxDiskAttachWaitSec = 10 * time.Second // Max time to wait for a disk to be attached
+	ntfs                 = "ntfs"
+	powershell           = "powershell"
+	diskNotFound         = "DiskNotFound"
+	diskpart             = "diskpart"
+	diskpartSuccessMsg   = "DiskPart successfully formatted the volume."
+
+	// Using a PowerShell script here due to lack of a functional Go WMI library.
+	// PowerShell script to identify the disk number given a scsi controller
+	// number and a unit number. The script returns an integer if the disk was
+	// found, else it returns DiskNotFound.
+	scsiAddrToDiskNumScript = `
+		Get-PnpDevice -FriendlyName 'VMware Virtual disk SCSI Disk Device' |
+		Select-Object -ExpandProperty InstanceId |
+		ForEach-Object {
+			$instanceId = $_
+			$uiNum = (
+				Get-PnpDeviceProperty -InstanceId $instanceId -KeyName 'DEVPKEY_Device_UINumber' |
+				Select -expand Data -erroraction 'silentlycontinue'
+			)
+			$addr = (
+				Get-PnpDeviceProperty -InstanceId $instanceId -KeyName 'DEVPKEY_Device_Address' |
+				Select -expand Data -erroraction 'silentlycontinue'
+			)
+			If ($uiNum -eq '%s' -And $addr -eq '%s') {
+				Write-Host ""
+				Get-WmiObject Win32_DiskDrive |
+				Where-Object { $_.PNPDeviceId -eq $instanceId } |
+				Select-Object -ExpandProperty Index
+				exit
+			}
+		}
+		Write-Host ""
+		Write-Host "DiskNotFound"
+	`
+
+	// DiskPart script to format a disk with a filesystem.
+	formatDiskScript = `
```

Comment:
@msterin I've created issue #1558 to add the build script, and #1559 for replacing the `diskpart` script.